packages/lesmis/src/components/MapOverlaysPanel/MapOverlayGroup.js

```diff
@@ -0,0 +1,368 @@
+    }
+
+    // Nested Label
+    .MuiFormLabel-root {
+      font-weight: normal;
+      font-size: 1rem;
+      line-height: 1.18rem;
+    }
+  }
+`;
+
+const Border = styled.div`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+
+  &:before {
+    ${VerticalDash};
+    top: 0;
+    left: -1.3rem;
+    height: ${DASH_OFFSET};
+  }
+
+  &:after {
+    ${VerticalDash};
+    top: ${DASH_OFFSET};
+    height: calc(100% - 1.81rem);
+    left: -1.3rem;
+  }
+
+  &.active {
+    &:after,
+    &:before {
+      ${VerticalHighlight};
+    }
+  }
+
+  &.selected {
+    &:before {
+      ${VerticalHighlight};
+    }
+  }
+`;
+
+const FormLabel = styled(MuiFormLabel)`
+  position: relative;
+  display: flex;
+  align-items: flex-start;
+  color: ${props => props.theme.palette.text.primary};
+  font-weight: 500;
+  font-size: 1rem;
+  line-height: 1.18rem;
+  padding-top: 1.25rem;
+  padding-right: 1rem;
+  padding-bottom: 1.25rem;
+
+  &.open:after {
+    ${VerticalDash};
+    top: 2.6rem;
+    height: calc(100% - 2.6rem);
+    left: 0.68rem;
+  }
+
+  &.selected {
+    &:after {
+      ${VerticalHighlight};
+    }
+  }
+
+  &:hover {
+    cursor: pointer;
+
+    .MuiSvgIcon-root {
+      color: ${props => props.theme.palette.text.secondary};
+    }
+  }
+
+  .MuiSvgIcon-root {
+    position: relative;
+    color: ${props => props.theme.palette.text.tertiary};
+    margin-right: 0.625rem;
+    margin-top: -0.3rem;
+    top: 2px;
+  }
+
+  .MuiSvgIcon-root.MuiSvgIcon-colorPrimary {
+    color: ${props => props.theme.palette.primary.main};
+  }
+
+  &.Mui-focused {
+    color: ${props => props.theme.palette.text.primary};
+  }
+`;
+
+const FormControlLabel = styled(MuiFormControlLabel)`
+  position: relative;
+  font-size: 0.875rem;
+  color: ${props => props.theme.palette.text.secondary};
+
+  .MuiRadio-root:before {
+    ${HorizontalDash};
+    top: 50%;
+    width: 1.125rem;
+    left: -0.56rem;
+  }
+
+  &:before {
+    ${VerticalDash};
+    top: 0;
+    height: 50%;
+    left: -0.625rem;
+  }
+
+  &:after {
+    ${VerticalDash};
+    top: 50%;
+    height: 50%;
+    left: -0.625rem;
+  }
+
+  &.active {
+    &:after,
+    &:before {
+      ${VerticalHighlight};
+    }
+  }
+
+  &.selected {
+    .MuiRadio-root:before {
+      ${HorizontalHighlight};
+    }
+
+    &:before {
+      ${VerticalHighlight};
+    }
+  }
+
+  &:last-child:after {
+    border: none;
+  }
+
+  &:hover {
+    cursor: pointer;
+
+    .MuiSvgIcon-root {
+      color: ${props => props.theme.palette.primary.main};
+    }
+  }
+`;
+
+const Radio = styled(MuiRadio)`
+  &.Mui-checked {
+    color: ${props => props.theme.palette.primary.main};
+
+    & + .MuiFormControlLabel-label {
+      color: ${props => props.theme.palette.primary.main};
+    }
+  }
+
+  .MuiSvgIcon-root {
+    font-size: 1.5rem;
+  }
+`;
+
+/**
+ * Calculates whether the map overlay group is the selected one or is part of the
+ * path to the selected one
+ */
+const getActiveClass = (path, selectedPath) => {
```

Comment:
My main suggestion out of this PR is to change the path to hold a list of codes rather than indexes. A couple of advantages to that:
- Allow easier recreation of selected path from a deep link
- Avoid any bugs where the components rerender and for some reason change order
- A little easier to interpret when debugging, if you print out the selected path

Fortunately, I don't think there is anything about your current setup that precludes that from happening! Hopefully it's just swapping out the setting of the path from using indexes to, ideally, `option.code`*, but if you want to avoid any back end changes, `option.name` for now. 

*The map overlay groups don't currently include their code, though it would be easy enough to change that and would be non-breaking on web-config-server. Also, the overlays use `measureId`, but I feel that could be renamed to just `code` on LESMIS.

Reply:
I agree that it would be better if possible to use a code rather than an index. However I started implementing it and although it was easy enough to add the code to the meditrak response it actually doesn't easily work with the algorithm to get the active class because we need to know where or not any given node is before or after the selected node to highlight the path down to it. It would work if the overlay list was alphabetical but it is not.

Reply:
Regarding the 3 points above. 
- I think the first one is a valid point, it could be done at the moment but it would be a bit awkward
- I think point 2 is quite hypothetical and at least as likely to happen with the codes
- I think point 3 is negligible difference